File: server/modules/auth/local.ts
import type { Context } from "hono";
import { getConnInfo } from "@hono/node-server/conninfo";

/** True for loopback TCP peers (same machine as the server). */
export function isLoopbackAddress(address: string | undefined): boolean {
  if (!address) return false;
  if (address === "127.0.0.1" || address === "::1" || address === "::ffff:127.0.0.1") {
    return true;
  }
  return address.startsWith("127.");
}

export function getClientAddress(c: Context): string {
  try {
    const info = getConnInfo(c);
    return info.remote?.address ?? "";
  } catch {
    return "";
  }
}

export function isLocalRequest(c: Context): boolean {
  return isLoopbackAddress(getClientAddress(c));
}

File: server-rs/src/auth/local.rs
/// True for loopback TCP peers (same machine as the server).
pub fn is_loopback_address(address: &str) -> bool {
    if address.is_empty() {
        return false;
    }
    if address == "127.0.0.1" || address == "::1" || address == "::ffff:127.0.0.1" {
        return true;
    }
    address.starts_with("127.")
}

/// Alias for checking whether a remote address is local to the server.
pub fn is_local_addr(address: &str) -> bool {
    is_loopback_address(address)
}
